crates/subspace-gateway/src/commands/run/dsn.rs

```diff
@@ -0,0 +1,99 @@
+//! DSN config and implementation for the gateway.
+
+use crate::node_client::{NodeClient, RpcNodeClient};
+use anyhow::anyhow;
+use clap::{Parser, ValueHint};
+use subspace_networking::libp2p::kad::Mode;
+use subspace_networking::libp2p::Multiaddr;
+use subspace_networking::{construct, Config, KademliaMode, Node, NodeRunner};
+use tracing::{debug, info};
+
+/// Configuration for network stack
+#[derive(Debug, Parser)]
+pub(crate) struct NetworkArgs {
+    /// WebSocket RPC URL of the Subspace node to connect to.
```

Comment:
`listen-on` is always helpful for local debugging as well as for setting up some cloud infrastructure. I'd add this parameter.